common/app/assets/DiscussionAssets.scala

```diff
@@ -0,0 +1,99 @@
+package assets
+
+import java.net.URI
+
+import app.LifecycleComponent
+import common.{AkkaAgent, ExecutionContexts, GuardianConfiguration, JobScheduler, Logging}
+import conf.switches.Switches
+import play.api.libs.ws.{WSClient, WSResponse}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+/**
+  * External assets are not hosted by frontend but referenced in an assets map
+  *
+  * This class pull the assets map regularly and translates a generic name into a full URL
+  *
+  * The map is a JSON object looking like
+  * {
+  *   "name": "js/name.min.hash.js"
+  * }
+  * The path in the object value is relative to the assets map
+  */
+class DiscussionExternalAssetsLifecycle (config: GuardianConfiguration, wsClient: WSClient, jobs: JobScheduler)
+  extends LifecycleComponent with ExecutionContexts with Logging {
+
+  def refresh(): Future[Unit] = {
+    config.discussion.frontendAssetsMap match {
+      case Some(url) if Switches.DiscussionFetchExternalAssets.isSwitchedOn =>
+        fetchAssetsMap(url, wsClient).map(
```

Comment:
I think that's the idea of this PR, that discussion is decoupled from and independent of frontend with a fairly defined API.  I'm assuming discussion-new will be deployed with riffraff also, so hopefully we would have enough information to know what was deployed at a given time.
Personally i feel like frontend is very big and would benefit from being split down.  However as you say, with this change it makes it harder to make changes across all parts of the code base and deploy together, but I think the interface between frontend and discussion is small enough to make sense.  Maybe there's another way to do it that keeps the atomic deployment though (red/green canary stack perhaps?)